API/Controllers/FileController.cs

```diff
@@ -0,0 +1,216 @@
+                string path = await fileUploader.UploadSingleFile(fileResource.File, newFileName);
+                User user = await HttpContext.GetContextUser(userService)
+                                             .ConfigureAwait(false);
+
+                File file = new File(path, newFileName, user, uploadDateTime);
+                fileService.UploadSingleFile(file);
+
+                return Ok(mapper.Map<File, FileResourceResult>(file));
+            } catch(FileExistException fileExistException)
+            {
+                ProblemDetails problem = new ProblemDetails
+                                         {
+                                             Title = fileExistException.Message,
+                                             Detail = "Please rename filename.",
+                                             Instance = "D902F8C6-23FF-4506-B272-C757BD709464"
+                };
+                return BadRequest(problem);
+            }
+        }
+
+        /// <summary>
+        /// Find file by id
+        /// </summary>
+        /// <param name="fileId"></param>
+        /// <returns> File </returns>
+        [HttpGet("{fileId}")]
+        public async Task<IActionResult> GetSingleFile(int fileId)
+        {
+            File file = await fileService.FindAsync(fileId);
+
+            if(file == null)
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "File could not be found.",
+                    Detail = "File could not be found.",
+                    Instance = "875B6402-D771-45EC-AB56-3DE0CDD446D6"
+                };
+                return NotFound(problem);
+            }
+
+            return Ok(file);
+        }
+
+        /// <summary>
+        /// Deletes single file
+        /// </summary>
+        /// <param name="fileId"></param>
+        /// <returns></returns>
+        [HttpDelete("{fileId}")]
+        [Authorize]
+        public async Task<IActionResult> DeleteSingleFile(int fileId)
+        {
+            File file = await fileService.FindAsync(fileId);
+            User user = await HttpContext.GetContextUser(userService)
+                                         .ConfigureAwait(false);
+
+            if(file == null)
+            {
+                ProblemDetails problem = new ProblemDetails
+                                         {
+                                             Title = "File was not found.",
+                                             Detail = "File was not found.",
+                                             Instance = "9D3830A2-E7D1-4610-A147-1D43BFB8DDBC"
+                };
+                return NotFound(problem);
+            }
+
+            if(!file.Uploader.Id.Equals(user.Id))
+            {
+                if(!userService.UserHasScope(user.IdentityId, Defaults.Roles.Administrator))
```

Comment:
We should really use scopes instead of role checks like these. Keeps things organized. Please create a new scope and check for that. You can refer to UserController.cs#UpdateAccount as an example.